err.go

```diff
@@ -15,24 +15,51 @@
 
 package ionhash
 
-import "fmt"
+import (
+	"fmt"
+
+	"github.com/amzn/ion-go/ion"
+)
 
 // An InvalidOperationError is returned when a method call is invalid for the struct's current state.
 type InvalidOperationError struct {
 	structName string
 	methodName string
+	message    string
 }
 
 func (e *InvalidOperationError) Error() string {
-	return fmt.Sprintf(`ionhash: invalid operation error on %v.%v`, e.structName, e.methodName)
+	if e.message != "" {
+		return fmt.Sprintf(`ionhash: %v.%v: %v`, e.structName, e.methodName, e.message)
+	} else {
```

Comment:
I've seen this code in other PRs. We should make a task for ourselves to go over such pieces of code once we have unit tests up and running to make sure we have consistent behaviour